EIPS/eip-2666.md

```diff
@@ -0,0 +1,107 @@
+---
+eip: 2666
+title: Repricing of precompiles and Keccak256 function
+author: Alex Vlasov (@shamatar)
+discussions-to: https://ethereum-magicians.org/t/eip2666-global-precompiles-repricing-and-many-more-discussion-thread/4332
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-05-22
+requires: 1352, 2046, 2565
+---
+
+## Simple Summary
+This EIP tries to fix the problems with Ethereum precompiles and built-in EVM function:
+- EIP-2046 changes a `STATICCALL` cost to precompile and it may be necessary to adjust costs of some precompiles that have taken old large cost (`700` gas) into account 
+- Some precompiles are overpriced and their pricing formulas do not reflect the structure of underlying function
+- Keccak256 built-in function in EVM has pricing that does not reflect underlying hash function structure
+- Better specify behavior of how gas handled if the precompile call fails (DDoS handling behavior)
+
+## Abstract
+Costs of many precompiles and built-in functions are invalid at the current state of the clients. This EIP contains a list of changes to the pricing formulas to better reflect underlying computations' structure.
+
+## Motivation
+Historic pricing for these functions in EVM does not reflect inner structure of the underlying computations (inner structure of the hash functions).
+
+## Specification
+
+If `block_number >= X` set the following prices to precompiles and Keccak256 opcode:
+  - `SHA256` precompile (address `0x02`) should be priced as `10 + ((len(input) + 8)/64 + 1) * 9`
+  - RIPEMD precompile (address `0x03`) should be priced as `6 + ((len(input) + 8)/64 + 1) * 12`
+  - Keccak256 built-in function should be priced as `13 + (len(input)/136 + 1)*15`
+
+This EIP *ideally* requires that `MODEXP` repricing is [implemented](https://eips.ethereum.org/EIPS/eip-2565) to also accurately reflect that there is implicit compensation for an old `STATICCALL` cost (pre-2046).
```

Comment:
```suggestion
This EIP *ideally* requires that `MODEXP` repricing is [implemented](./eip-2565.md) to also accurately reflect that there is implicit compensation for an old `STATICCALL` cost (pre-2046).
```

Reply:
Also, all opcodes must be in the form `STATICCALL (0xfa)` the first time they are referenced in the EIP at least (optional for further references).  I think we should do something similar for precompiles as well.